extensions/wikia/Wall/WallHistoryController.class.php

```diff
@@ -217,8 +219,18 @@ private function getFormatedHistoryData( $history, $threadId = 0 ) {
 				$history[$key]['msgurl'] = $messagePageUrl;
 
 				$msgUser = $wm->getUser();
+				$msgPage = Title::newFromText( $msgUser->getName(), $ns );
+				if ( empty( $msgPage ) ) {
```

Comment:
Yup, the message causing the error in this case, is from a ```$msgUser``` who is an anonymous user.  For some reason, someone ran a Special:Block on this IP address an included the HideUser option which hides user attribution (which doesn't make much sense for an IP).  This is what I suspect anyway, because the ```revision``` row loaded has the ```DELETED_USER``` flag set on it.

The call to ```$msgUser->getName()``` therefore returns an empty string rather than the IP address an anonymous user usually would have because of the HideUser restriction.  This means an empty "title" is used as an argument to ```Title::newFromText``` which returns an empty object.

I spent about 2.5 hours getting as close as I could to the root cause.  The only errors in production are for this wiki, and for this exact thread ID:

https://kibana.wikia-inc.com/index.html#dashboard/temp/AUyVOsKZrWhzbAyo4bmn

 Given this, and that most of the 138 instances of this error over the last month are likely from people at Wikia testing that URL, I can't say that this was time well spent.